helix-rest/src/main/java/org/apache/helix/rest/server/HelixRestObjectNameFactory.java

```diff
@@ -0,0 +1,86 @@
+package org.apache.helix.rest.server;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.Hashtable;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+
+import com.codahale.metrics.jmx.ObjectNameFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Creates an {@link ObjectName} that has "name", "type", "namespace" properties
+ * for metrics registry in Helix rest service.
+ *
+ * <p>It is recommended to only be used within Helix REST.
+ */
+class HelixRestObjectNameFactory implements ObjectNameFactory {
+  private static final Logger LOG = LoggerFactory.getLogger(HelixRestObjectNameFactory.class);
+
+  private static final String KEY_NAME = "name";
+  private static final String KEY_TYPE = "type";
+  private static final String KEY_NAMESPACE = "namespace";
+
+  public ObjectName createName(String type, String domainNameSpace, String name) {
+    String[] splits = domainNameSpace.split(":");
+    String domain = splits[0];
+    String namespace = splits[1];
+
+    try {
+      Hashtable<String, String> properties = new Hashtable<>();
+      properties.put(KEY_NAME, name);
+      properties.put(KEY_TYPE, type);
+      properties.put(KEY_NAMESPACE, namespace);
+
+      /*
+       * The only way we can find out if we need to quote the properties is by
```

Comment:
Given this is an internally used class, do we expect any value that needs quote?

Reply:
I think it doesn't hurt to have the quote check, as it supports regex query in object name: `new ObjectName("helix*")`

Reply:
I don't think it hurts to have the quote check. Eg. if we use regex, quote is needed: `new ObjectName("helix*")`

Reply:
I personally prefer less protection so the logic fails earlier during test.
But no strong preference here.

Reply:
I will keep it then.